backend/emitaux.ml

```diff
@@ -209,6 +212,41 @@ type emit_frame_actions =
   }
 
 let emit_frames a =
+  let emit_u8 n =
+    if n < 0 || n > 0xFF
+    then
+      Misc.fatal_errorf
+        "attempting to emit unsigned 8-bit integer %d out of range" n
+    else a.efa_u8 (Numbers.Uint8.of_nonnegative_int_exn n)
+  in
+  let emit_i16 n =
+    if n < -0x8000 || n > 0x7FFF
+    then
+      Misc.fatal_errorf
+        "attempting to emit signed 16-bit integer %d out of range" n
+    else a.efa_i16 (Numbers.Int16.of_int_exn n)
+  in
+  let emit_u16 n =
+    if n < 0 || n > 0xFFFF
+    then
+      Misc.fatal_errorf
+        "attempting to emit unsigned 16-bit integer %d out of range" n
+    else a.efa_u16 (Numbers.Uint16.of_nonnegative_int_exn n)
```

Comment:
`of_int_exn` and `of_nonnegative_int_exn` in `Numbers` already has this check, why having another one here?